### DISCUSSION TITLE
How to obtain volume-averaged concentration on PyBamm.

### DISCUSSION THREAD

**Discussion by HassanSewailem** [pybamm-team/PyBaMM#2604] [Q&A] [answered]
Dear PyBamm team, 

I am trying to obtain the volume-averaged value for the spatial variable 'Negative particle surface concentration [mol.m-3]' for the DFN and SPMe models. Any thoughts on how to obtain this? 

Best regards,
Hassan

**Comment by valentinsulzer** [comment, marked as answer]
"X-averaged negative particle surface concentration [mol.m-3]".

You can use `variables.search` to find list of available variables e.g. `model.variables.search("surface concentration")`
